jazzhands/__main__.py

```diff
@@ -58,73 +87,109 @@ def process_jsx(root):
     else:
         _process_jsx(root)
 
+
 def collect_app_asset_src(dirs, lang):
+    """Collects (copy or symlink) assets from apps into the project.
+
+    JavaScript "packages" are placed in node_modules/ to be imported. Less and
+    Stylus packages are placed adjacent to the projects own index.less or
+    index.styl for relative importing.
+
+    If the file is Javascript it is also processed for any necessary JSX/ES
+    transpiling after being installed in node_modules/
+    """
+
+    if lang not in dirs:
+        print("No %s to collect" % (lang,))
+        return
     for (app_name, app_asset_dir) in dirs[lang]:
         dest_dir = os.path.join(os.path.dirname(index_files[lang]), app_name)
-        
+
         print("collecting assets", app_name, lang, app_asset_dir, '->', dest_dir)
         if lang == 'js':
             dest_dir = os.path.join("node_modules", app_name)
             for root, dirs, files in os.walk(app_asset_dir):
                 for fn in files:
                     if fn.endswith('.js'):
-                        js_watch[os.path.join(root, fn)] = os.stat(os.path.join(root, fn)).st_mtime
+                        watch['js'][os.path.join(root, fn)] = os.stat(os.path.join(root, fn)).st_mtime
 
         pull_app_assets(app_asset_dir, dest_dir, copy=(lang == 'js'))
 
         if lang == 'js':
             process_jsx(dest_dir)
 
+
 def build_stylus(dirs):
     print("Building Stylus")
     in_file = open(index_files['styl'], 'r')
-    out_file = open(os.path.join(css_dir, 'bundle.css'), 'w')
+    out_file = open(os.path.join(css_dir, 'bundle.css'), 'a')
     stylus_dir = os.path.dirname(index_files['styl'])
     stylus_bin = os.path.join(os.path.relpath(".", stylus_dir), "node_modules/.bin/stylus")
+    if not os.path.exists(stylus_bin):
+        print("ERROR: Stylus files found, but stylus is not installed.")
+        print("To fix, install stylus in this project using NPM:")
+        print("    npm install --save stylus")
+        print("Or, using Yarn:")
+        print("    yarn add stylus")
+        sys.exit(1)
     subprocess.call([stylus_bin, '--resolve-url'], cwd=stylus_dir, stdin=in_file, stdout=out_file)
 
+
 def build_less(dirs):
     print("Building Less")
     subprocess.call(['lessc', index_files['less'], os.path.join(css_dir, 'bundle.css')])
 
+
 def build_js(dirs):
     print("Building JS")
     args = ['./node_modules/.bin/browserify']
-    # args.extend("-t [ babelify --presets [ react es2015 ] ]".split())
+    args.extend("-t [ babelify --presets [ react es2015 ] ]".split())
     args.extend([index_files['js'], '-o', os.path.join(js_dir, 'bundle.js')])
-    p = subprocess.Popen(args, stderr=subprocess.PIPE)
-    out, err = p.communicate()
+    proc = subprocess.Popen(args, stderr=subprocess.PIPE)
+    out, err = proc.communicate()
 
-    if p.returncode > 0:
+    if proc.returncode > 0:
+        # Experimental "auto install" feature for missing NPM dependencies during development
```

Comment:
As long as ```--auto-npm``` is here, could we mention it in the README?  Also, any reason not to install all the missing dependencies in one go?

Reply:
It causes an error on the first one it finds, and after installing that it can include other dependencies so the results could change and we should try again to get current results.

I'll document the flag.